### PR TITLE
add keep to users where appropriate

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim @andrewconner 

adding participants to a created thread doesn't create a KTU, neither does commenting on a keep. this should do it.

@ryanpbrewster I think we should create a specific method for updating KTU + keep.connections, since we're losing `addedBy` here. It'd be similar to `persistKeep` but only for updating users, and asking for `addedBy`.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 1]
```diff
@@ -663,13 +664,23 @@ class KeepCommanderImpl @Inject() (
...
+  def addUsersToKeep(keepId: Id[Keep], addedBy: Id[User], newUsers: Set[Id[User]])(implicit session: RWSession): Keep = {
+    val k = keepRepo.get(keepId)
+    require(k.connections.users.contains(k.userId), "keep owner is not one of the connected users")
+
+    val keep = keepRepo.save(k.withParticipants(k.connections.users ++ newUsers))
+
+    newUsers.foreach { userId => ktuCommander.internKeepInUser(keep, userId, addedBy) }
+
+    keep
+  }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:681]: @ryanpbrewster please let me know what you think here. I have no idea what `addedBy` is used for, but if we ever decide to use it we can't use `persistKeep` as is.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala [hunk 2]
```diff
@@ -44,7 +44,11 @@ class DiscussionCommanderImpl @Inject() (
     }
 
     errs.headOption.map(fail => Future.failed(fail)).getOrElse {
-      eliza.sendMessageOnKeep(userId, text, keepId)
+      val message = eliza.sendMessageOnKeep(userId, text, keepId)
+      db.readWrite { implicit s =>
+        keepCommander.addUsersToKeep(keepId, userId, Set(userId))
+      }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala:49]: Maybe add them before having Eliza send the message.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 3]
```diff
@@ -663,13 +664,23 @@ class KeepCommanderImpl @Inject() (
...
+    keep.connections.users.foreach { userId => ktuCommander.internKeepInUser(keep, userId, userId) }
 
     val libraries = libraryRepo.getActiveByIds(keep.connections.libraries).values
     libraries.foreach { lib => ktlCommander.internKeepInLibrary(keep, lib, keep.userId) }
 
     keep
   }
+  def addUsersToKeep(keepId: Id[Keep], addedBy: Id[User], newUsers: Set[Id[User]])(implicit session: RWSession): Keep = {
+    val k = keepRepo.get(keepId)
+    require(k.connections.users.contains(k.userId), "keep owner is not one of the connected users")
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: Why?

**Comment by ryanpbrewster** [comment]
lgtm, I think it's a good change.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala:49]: Sounds good, I was thinking about performance but better to not introduce a potential race condition.

**Pull request merged**